sdk/core/core-https/src/nodeHttpsClient.ts

```diff
@@ -198,9 +198,17 @@ class NodeHttpsClient implements HttpsClient {
     const proxySettings = request.proxySettings;
     if (proxySettings) {
       if (!this.proxyAgent) {
+        let parsedUrl: URL;
+        try {
+          parsedUrl = new URL(proxySettings.host);
+        } catch (_error) {
+          throw new RangeError("Expecting a valid host string in proxy settings.");
```

Comment:
Can we include what the bad value is for easier debug?

```suggestion
          throw new RangeError(`Expecting a valid host string in proxy settings, but found "${proxySettings.host}".`);
```